File: estruturas/morris/morris.py
from random import randint


class Morris:
    def __init__(self):
        self.X = 0

    def adiciona(self):
        r = randint(0, 2 ** self.X - 1)
        if r == 0:
            self.X = self.X + 1

    def conta(self):
        return 2 ** self.X - 1

File: estruturas/morris/c++/morris.cpp
#include "morris.hpp"
#include <cstdlib>

Morris::Morris() {
  X = 0;
}

void Morris::adiciona() {
  int r = rand() % (1 << X);
  if (r == 0) X++;
}

int Morris::conta() {
  return (1 << X) - 1;
}